File: frontend/src/types/booking.ts
export type BookingStatus = 'Pending' | 'Confirmed' | 'Completed' | 'Cancelled';

export interface AvailableSlot {
  startTime: string;   // ISO UTC format
  endTime: string;     // ISO UTC format
}

export interface AvailableSlotsQueryParameters {
  serviceId: string;
  staffId: string;
  date: string;        // yyyy-MM-dd
}

export interface Booking {
  id: string;
  bookingCode: string;
  customerId: string;
  customerName?: string;
  customerEmail?: string;
  serviceId: string;
  serviceName?: string;
  servicePrice?: number;
  durationMinutes?: number;
  staffId: string;
  staffName?: string;
  startTime: string;   // ISO UTC
  endTime: string;     // ISO UTC
  status: BookingStatus;
  customerNote?: string | null;
  cancellationReason?: string | null;
  createdAt: string;
}

export interface CreateBookingRequest {
  serviceId: string;
  staffId: string;
  startTime: string;   // ISO UTC
  customerNote?: string;
}

export interface UpdateBookingStatusRequest {
  status: BookingStatus;
  cancellationReason?: string;
}

export interface CancelBookingRequest {
  cancellationReason: string;
}

export interface BookingQueryParameters {
  date?: string;
  status?: BookingStatus;
  staffId?: string;
  search?: string;
  page?: number;
  pageSize?: number;
}

export interface MyBookingQueryParameters {
  date?: string;
  status?: BookingStatus;
  page?: number;
  pageSize?: number;
}

File: ServiceBooking.Api/Models/Booking.cs
using ServiceBooking.Api.Common;

namespace ServiceBooking.Api.Models;

public class Booking
{
    public Guid Id { get; set; } = Guid.NewGuid();
    public string BookingCode { get; set; } = string.Empty;
    public Guid CustomerId { get; set; }
    public Guid ServiceId { get; set; }
    public Guid StaffId { get; set; }
    public DateTime StartTime { get; set; }
    public DateTime EndTime { get; set; }
    public string Status { get; set; } = BookingStatus.Pending;
    public string? CustomerNote { get; set; }
    public string? CancellationReason { get; set; }
    public DateTime CreatedAt { get; set; } = DateTime.UtcNow;

    // Navigation properties
    public User Customer { get; set; } = null!;
    public Service Service { get; set; } = null!;
    public Staff Staff { get; set; } = null!;
}
